CodeGen/Generators/UnitsNetGen/QuantityGenerator.cs

```diff
@@ -792,7 +794,8 @@ public bool Equals({_quantity.Name} other, double tolerance, ComparisonType comp
         /// <returns>A hash code for the current {_quantity.Name}.</returns>
         public override int GetHashCode()
         {{
-            return new {{ QuantityType, Value, Unit }}.GetHashCode();
+            var roundedBaseValue = Math.Round(GetValueInBaseUnit(), 5);
+            return new {{ QuantityType, roundedBaseValue }}.GetHashCode();
```

Comment:
This deserves some comments explaining the rationale for the GetHashCode. It is not intuitive to me, had I not known about the discussion.

Why is `5` a good choice here?

Reply:
QuantityType was added as you could have two different units (say Length and Area) with identical values. If they also had the same unit enum value (treated as int), they would have the same hash code.

Reply:
Is it problematic that different quantity types would share hash code though? They would still not be equal.
As long two equal values produce the same hash code, we should be fine. There is no requirement that in-equal values should produce different hash codes, as mentioned in https://github.com/angularsen/UnitsNet/pull/838#discussion_r492287743.

Reply:
Honestly- it's an over-estimation of the max rounding error that we could get by converting to base (it's so stated in the Precision chapter of the wiki).

Reply:
https://github.com/angularsen/UnitsNet/issues/717#issuecomment-548425378
> HashSet may have been a better example - for set operations. It too supports an IEqualityComparer to be passed in.

So one impact would be that
```c#
var hs = new HashSet<IQuantity>();
hs.Add(Mass.FromKilograms(1));
hs.Add(Mass.FromKilograms(1));
hs.Add(Length.FromMeters(1));
hs.Dump();

/* Today
5HashSet<IQuantity> (2 items)
--
1 kg
1 m
*/

/* This PR would change it to this?
5HashSet<IQuantity> (2 items)
--
1 m
*/

```

Reply:
Ok I'm not sure what I'm reading but QuantityValue is still there hah. Ignore that.

Just for our own learning: it does work! I thought it would disallow it.
The difference is without the quantity type it has to call Equals to check if it's a real collision or not. Given how easy it is to avoid, without a significant hash cost, it should be included for performance.

Reply:
By the way (I didn't know that) but renaming any of the variables forming the anonymous class `new { QuantityType, roundedBaseValue }`, would produce a wholly different result from the GetHashCode that follows it :)

Reply:
> Ok I'm not sure what I'm reading but QuantityValue is still there hah. Ignore that.

LOL! I read it that way too! 😆